nav2_amcl/src/amcl_node.cpp

```diff
@@ -1543,6 +1553,10 @@ AmclNode::initServices()
     "reinitialize_global_localization",
     std::bind(&AmclNode::globalLocalizationCallback, this, _1, _2, _3));
 
+  initial_guess_srv_ = create_service<nav2_msgs::srv::SetInitialPose>(
```

Comment:
This must be reset on cleanup like the others

Reply:
Ah yes, I missed that. I added calling the reset function on cleanup like the others